docs/source/user-guide/guide_file_fetchers.rst

```diff
@@ -0,0 +1,57 @@
+Implementing a custom file fetcher
+----------------------------------
+
+DKAN uses a library called getdkan/file-fetcher. This library allows developers to extend the file transfer functionality for their specialized needs.
+
+This library is used to download a resource, such as a CSV file, so that it can be loaded into the database and presented through the UI and API. This process is called 'localization,' because the source resource is copied to the local file system.
+
+The standard file fetcher processors will probably be adequate for most uses, but there could be other use cases, such as needing to authenticate, or getting a file from S3 instead of HTTP.
```

Comment:
Let's be consistent about the canonical name of "file-fetcher" - I'm not sure I like the all-lowercase-with-hyphen styling but that's what it is in the repo README for now.

```suggestion
The standard file-fetcher processors will probably be adequate for most uses, but there could be other use cases, such as needing to authenticate, or getting a file from S3 instead of HTTP.
```